ekm/key_manager.go

```diff
@@ -0,0 +1,38 @@
+package ekm
+
+import (
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	ssz "github.com/ferranbt/fastssz"
+	spectypes "github.com/ssvlabs/ssv-spec/types"
+)
+
+type KeyManager interface {
+	spectypes.BeaconSigner
+	SlashingProtector
+	// AddShare decrypts and saves an encrypted share private key
+	AddShare(encryptedSharePrivKey []byte, sharePubKey phase0.BLSPubKey) error
+	// RemoveShare removes a share key
+	RemoveShare(pubKey phase0.BLSPubKey) error
+}
+
+type ShareDecryptionError error
+
+type BlockRootWithSlot struct {
+	BlockRoot phase0.Root
+	Slot      phase0.Slot // Slot is just a payload, so BlockRootWithSlot's methods don't use it
+}
+
+func (b BlockRootWithSlot) HashTreeRoot() ([32]byte, error) {
+	return ssz.HashWithDefaultHasher(spectypes.SSZBytes(b.BlockRoot[:]))
+}
+
+func (b BlockRootWithSlot) GetTree() (*ssz.Node, error) {
+	return ssz.ProofTree(spectypes.SSZBytes(b.BlockRoot[:]))
+}
+
+func (b BlockRootWithSlot) HashTreeRootWith(hh ssz.HashWalker) error {
+	indx := hh.Index()
+	hh.PutBytes(b.BlockRoot[:])
+	hh.Merkleize(indx)
+	return nil
+}
```

Comment:
if we don't expect these funcs to be called (because its wrong), then why not return errors instead to avoid supporting wrong behavior?

Reply:
or we could alternatively make them right by generating the code (which would include Slot)

Reply:
Why don't we? We call `HashTreeRoot` in `ComputeETHSigningRoot` and it needs to be calculated for `BlockRoot` only

Reply:
![image](https://github.com/user-attachments/assets/10a33942-49ac-46f5-a869-ae71a8d933fe)
the spec already has a type to wrap block roots, we can just add this line

Reply:
I had to add a slot to `SignBeaconObject` method of `BeaconSinger`'s interface because of https://github.com/ssvlabs/ssv/pull/2028#discussion_r1989548175, so we don't need `BlockRootWithSlot` anymore, I removed it. So I'm marking this as resolved